sdk/core/core-http/src/fetchHttpClient.ts

```diff
@@ -66,6 +69,28 @@ export abstract class FetchHttpClient implements HttpClient {
         }
       };
       httpRequest.abortSignal.addEventListener("abort", abortListener);
+
+      registerCleanUpListenerOnStream = (stream: NodeJS.ReadableStream, isDownload: boolean) => {
```

Comment:
instead of using a boolean `isDownload`, what about something like this:

```suggestion
      registerCleanUpListenerOnStream = (stream: NodeJS.ReadableStream, streamType: "download" | "upload") => {
```

It seems like it would be more readable, since you wouldn't be passing true/false to the method when calling it